src/repositories/message.js

```diff
@@ -0,0 +1,47 @@
+const errors = require("../utils/error/errors");
+module.exports = [
+    "msgModel",
+    (msgModel) => {
+        return{
+            create: async (opt) => {
+                const newMessage = new msgModel(opt.query);
+                await newMessage.save();
+
+                return newMessage;
+            },
+
+            get:async (opt) => {
+                const message = await msgModel.findOne(opt.query);
+
+                if (!message) {
+                    throw new Error(errors.MESSAGE_NOT_EXIST);
+                  }
+
+                return message;
+
+            },
+
+            getMessageForRoom: async (opt) => {
+                
+            },
+
+            delete: async(opt) => {
+                const message = await msgModel.findOne(opt.query);
+                await message.remove();
+                return message;
+            },
+
+            update: async(opt) =>{
+                const message = await msgModel.findOne(opt.query);
```

Comment:
you can just use your get function it does this for you already

Reply:
yes

Reply:
with how im defining it, its out of scope

Reply:
done